tests/std/tests/P0429R9_flat_map/test.cpp

```diff
@@ -72,6 +72,44 @@ bool check_value_content(const T& obj, const typename T::mapped_container_type&
     return ranges::equal(actual, expected);
 }
 
+enum class subrange_type : bool {
+    equal,
+    permutation,
+};
+
+struct subrange_t {
+    ptrdiff_t first_index;
+    ptrdiff_t last_index;
+    subrange_type type;
+};
+
+using subranges_t = std::vector<subrange_t>;
+
+template <IsFlatMap T>
+bool check_value_content(
+    const T& obj, const typename T::mapped_container_type& expected, const subranges_t& subranges) {
+    const auto& actual = obj.values();
+    if (actual.size() != expected.size()) {

```

Comment:
Pre-existing: This is good and necessary, but the previous `check_key_content` and `check_value_content` are doing this unnecessarily.